### PR TITLE
Support isWrapperFor and unwrap

### PR DESCRIPTION
**Pull request by gabrywu**
### _Why are the changes needed?_
support `isWrapperFor` and `unwrap` to make KyuubiConnection easy to use

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Comment by gabrywu** [comment]
@pan3793 Should I resolve above failed checks?

**Comment by pan3793** [comment]
> @pan3793 Should I resolve above failed checks?

Yes, I guess you missing the import

**Comment by gabrywu** [comment]
> > @pan3793 Should I resolve above failed checks?
> 
> Yes, I guess you missed the import

How to resolve it? I create a test based on current test class, so, I think I don't miss the import

**Comment by turboFei** [comment]
> > > @pan3793 Should I resolve above failed checks?
> > 
> > 
> > Yes, I guess you missed the import
> 
> How to resolve it? I create a test based on current test class, so, I think I don't miss the import

```
import org.apache.kyuubi.jdbc.hive.KyuubiConnection
```

The package of `KyuubiHiveDriverSuite` is  `org.apache.kyuubi.jdbc`

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java [hunk 1]
```diff
@@ -1585,8 +1585,7 @@ public void setTypeMap(Map<String, Class<?>> map) throws SQLException {
 
   @Override
   public boolean isWrapperFor(Class<?> iface) throws SQLException {
-    // TODO Auto-generated method stub
-    throw new SQLFeatureNotSupportedException("Method not supported");
+    return iface.isInstance(this);
```
Comment (turboFei) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java:1588]: what is the use case?

The implementation in this pr just likes isInstance and asInstance.

**Comment by gabrywu** [comment]
> > > @pan3793 Should I resolve above failed checks?
> > 
> > 
> > Yes, I guess you missed the import
> 
> How to resolve it? I create a test based on current test class, so, I think I don't miss the import

Sorry, I really missed the import, I will fix it

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java (hunk 1 above)
Comment (gabrywu) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java:1588]: I think it's a standard method, no matter whether it looks like isInstance and asInstance or not.

**Comment by pan3793** [comment]
Thanks, merging to master

**Comment by yaooqinn** [comment]
thanks @gabrywu for the first contribution and @pan3793 @turboFei for the review. PR assigned.

**Pull request closed without merging**